sklearn/metrics/_pairwise_distances_reduction.pyx

```diff
@@ -764,11 +1160,11 @@ cdef class PairwiseDistancesArgKmin(PairwiseDistancesReduction):
                     Y_start + j,
                 )
 
-    @final
```

Comment:
Does removing the `@final` here lead to any performance regressions for metrics that is not `euclidean `?

Reply:
Thanks for the comment. This needs to be assessed.

IIRC from my experiments, optimisations provided by `@final` mainly comes on classes that do not make use of polymorphism (but we do use polymorphism here).

It's more a matter of implementation explicitness: the usage of `@final` on methods indicates that this method is not overwritten in any subclass (similarly the usage of `@final` on classes indicates that those classes aren't extended).

In overall, I think we must check for any performance regression before merging this PR.